FlightStreamDeck.SimConnectFSX/SimConnectFlightConnector.cs

```diff
@@ -830,9 +838,9 @@ private void RegisterGenericValues()
 
             Task.Run(async () =>
             {
+                await smGeneric.WaitAsync();
```

Comment:
Can I check why you move this outside of the try block? Afaik, it has to be inside try block to ensure Release is executed on failure.

Reply:
It's technically possible for WaitAsync to fail, in which case you wouldn't want to attempt to release a semaphore that wasn't successfully acquired. Waiting right before the try block ensures that the finally block only runs if the wait succeeds.

Reply:
Yeah I was totally out of my mind 😂.